python/packages/nisar/workflows/nisar_l0b_dm2_to_dbf.py

```diff
@@ -484,22 +578,49 @@ def nisar_l0b_dm2_to_dbf(args):
                 # mid AZ time at the center of the AZ block
                 azt_mid = azt_raw[rgl_slice].mean()
 
+                # Compute RX ANT Power Pattern calibration if requested
+                if args.rx_antpat_calib:
+                    if plot:
+                        plot_name = plot_name = out_path.joinpath(
+                            f'Plot_OneTap_DBF_EL_RxAntPat_Freq{freq_band}_'
+                            f'Pol{txrx_pol}_AzBlock{n_blk}.png'
+                        )
+                    else:
+                        plot_name = None
+                    inv_rxpat_1w = _reconstruct_inverse_el_magpat_full_swath(
+                        el_cuts, el_points, az_trans, azt_mid, sr, orbit,
+                        attitude, dem, max_p2p_ant=args.max_p2p_ant,
+                        plot_name=plot_name)
+                else:  # No RX ANT Pattern Correction
+                    inv_rxpat_1w = None
+
                 if args.no_rgcomp:  # simply perform mosaicking
-                    echo_dbf = dbf_onetap_from_dm2(
+                    echo_dbf, rgb_limits = dbf_onetap_from_dm2(
                         dset_azblk[:, :num_rgl], azt_mid, el_trans, az_trans,
                         sr, orbit, attitude, dem, cal_coefs=amp_cal
                     )
+                    # Apply the inverse of RX EL amplitude pattern to the echo
+                    # if requested
+                    if args.rx_antpat_calib:
+                        # XXX convolve square of `inv_rxpat_1w` by pulsewidth
+                        # prior to echo multiplication to include pulse
+                        # extension!
+                        # Alternatively, apply ANT correction after range comp
+                        # as part of RSLC workflow (preferred)!
+                        inv_rxpat_1w[...] = fftconvolve(
+                            inv_rxpat_1w**2,
+                            (1 / nrgb_pw) * np.ones(nrgb_pw),
+                            mode='full'
+                            )[:sr.size]
+                        echo_dbf *= np.sqrt(inv_rxpat_1w)
```

Comment:
This seems a little strange to me. Wouldn't you want to smooth the pattern before taking the inverse? It's also confusing to change the units of a variable. I'm thinking something like
```suggestion
                        inv_rxpat_1w[...] = np.sqrt(1.0 / fftconvolve(
                            1.0 / inv_rxpat_1w**2,
                            (1 / nrgb_pw) * np.ones(nrgb_pw),
                            mode='full'
                            ))[:sr.size]
                        echo_dbf *= inv_rxpat_1w
```

Reply:
Yes, but given the inverse is already generated, the convolution is lazily and directly applied to the inverse. I can change it but I have not noticed any remarkable difference in the respective RSLCs using either ones except for an overall scalar.